src/adagrams.js

```diff
@@ -1,15 +1,95 @@
+import { letterPool, scoreChart } from 'constants';
+
+// helper function to create letter pool array
+export const createLetterPool = () => {
+  const letterPoolArray = [];
+
+  for (const letter in letterPool) {
+    for (let i = 0; i < letterPool[letter]; ++i) {
+      letterPoolArray.push(letter);
+    }
+  }
+
+  return letterPoolArray;
+};
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const letterPool = createLetterPool();
+  const playerHand = [];
+  let counter = 0;
+
+  while (counter < 10) {
+    // access a letter via random index from letter pool
+    const randomIndex = Math.floor(Math.random() * (letterPool.length - 1));
+    const randomLetter = letterPool[randomIndex];
+
+    // add letter to player hand, remove letter from letter pool
+    playerHand.push(randomLetter);
+    letterPool.splice(randomIndex, randomIndex);
+    counter += 1;
+  }
+
+  return playerHand;
 };
 
-export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+export const usesAvailableLetters = (inputWord, lettersInHand) => {
+  // check that each character in inputWord exists in lettersInHand
+  for (const char of inputWord) {
```

Comment:
We chatted about this in our 1-1: we could reduce the time complexity from O(n^2) to O(n) if we calculate frequency maps first, then compare those values.